src/main/java/io/lighty/yang/validator/formats/JsTree.java

```diff
@@ -61,23 +60,23 @@ public void emitFormat() {
         for (final RevisionSourceIdentifier source : this.sources) {
             final Module module = this.schemaContext.findModule(source.getName(), source.getRevision())
                     .orElseThrow(() -> new NotFoundException("Module", source.getName()));
-            final List<Integer> removeChoiceQnames = new ArrayList<>();
-            SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
+            final SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
 
+            final SchemaInferenceStack schemaInferenceStack = SchemaInferenceStack.of(this.schemaContext);
             // Nodes
-            printLines(getChildNodesLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getChildNodesLines(singletonListInitializer, module, schemaInferenceStack));
 
             // Augmentations
-            for (AugmentationSchemaNode augNode : module.getAugmentations()) {
+            for (final AugmentationSchemaNode augNode : module.getAugmentations()) {
                 printLines(getAugmentationNodesLines(singletonListInitializer.getSingletonListWithIncreasedValue(),
-                        augNode, removeChoiceQnames));
+                        augNode, schemaInferenceStack));
             }
 
             // Rpcs
-            printLines(getRpcsLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getRpcsLines(singletonListInitializer, module, schemaInferenceStack));
 
             // Notifications
-            printLines(getNotificationsLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getNotificationsLines(singletonListInitializer, module, schemaInferenceStack));
```

Comment:
Notifications have absolute patch consisting just of its QName. We do not need stack.

Reply:
Yes, but Notification can have child nodes, which is required to track with schemaIS.

Reply:
OK